ops/testing.py

```diff
@@ -361,52 +392,69 @@ def relation_set(self, relation_id, key, value, is_app):
         if bucket_key not in relation:
             relation[bucket_key] = {}
         bucket = relation[bucket_key]
-        bucket[key] = value
+        if value == '':
+            bucket.pop(key, None)
+        else:
+            bucket[key] = value
 
```

Comment:
Is this change related to this branch?

Reply:
There is a test added related to this change (test_relation_set deletes)
It was something that I encountered as part of working on test_model.py
This branch was a collection of changes to Harness that were necessary to do other testing, but they weren't independently factored apart.

Reply:
Ack